proto/cosmos/authz/v1beta1/authz.proto

```diff
@@ -6,22 +6,21 @@ import "google/protobuf/timestamp.proto";
 import "gogoproto/gogo.proto";
 import "google/protobuf/any.proto";
 
-option go_package = "github.com/cosmos/cosmos-sdk/x/authz/types";
+option go_package = "github.com/cosmos/cosmos-sdk/x/authz";
+option (gogoproto.goproto_getters_all) = false;
 
 // GenericAuthorization gives the grantee unrestricted permissions to execute
 // the provided method on behalf of the granter's account.
 message GenericAuthorization {
   option (cosmos_proto.implements_interface) = "Authorization";
 
-  // method name to grant unrestricted permissions to execute
-  // Note: MethodName() is already a method on `GenericAuthorization` type,
-  // we need some custom naming here so using `MessageName`
-  string method_name = 1 [(gogoproto.customname) = "MessageName"];
+  // Msg, identified by it's type URL, to grant unrestricted permissions to execute
+  string msg = 1;
 }
 
-// AuthorizationGrant gives permissions to execute
+// Grant gives permissions to execute
 // the provide method with expiration time.
-message AuthorizationGrant {
+message Grant {
   google.protobuf.Any       authorization = 1 [(cosmos_proto.accepts_interface) = "Authorization"];
   google.protobuf.Timestamp expiration    = 2 [(gogoproto.stdtime) = true, (gogoproto.nullable) = false];
 }
```

Comment:
But then we will need to move `expiration` to all `Authorizaiton` concrete types. It's possible - but to keep that requirement, we will probably need to extend the `Authorization` interface by `ExpiresAt() time.Time`.

Reply:
As per a Discord discussion, here are the pros and cons I can think of (ordered by importance, according to me).

**Pros of changing to 1 proto message**
  - Simplicity.
    - 1 grant type == 1 proto message, no nested stuff.
  - Centralized logic.
    - Everything happens in `Accept()`. There's no other magic (say pruning based on expiration date) that the user might need to read docs/code to know.
  - Possibility to have grants without expiration date (semantically).
  - State size is smaller.

**Cons of changing to 1 proto message**
  - We want to release `beta` asap. This change might need another ADR round.
  - No way to add a secondary index for expiration, which would allow pruning based on expiration.
  - Maybe slightly more code for grant creators: expiration logic is duplicated in each grant (though can be easily factorized into a function).

Reply:
> Centralized authorization logic.
> Everything happens in Accept(). There's no other magic (say pruning based on expiration date) that the user might need to read docs/code to know.
> Possibility to have grants without expiration date (without a expiration = nil hack).

These seem like cons to me.

I would also add:

**Cons:**
* client developers can't write user interfaces that show `expiration` as a top-level field for authorizations
* every `Authorization` needs to repeat `expiration` logic
* some `Authorization` types might accidentally be missing `expiration` as an option

Reply:
In general, I don't see a lot of value in moving something which can be generalized up into a polymorphic `Any` type, especially with protobuf. We do this in `x/authz` and now clients need to switch on every `Account` interface type just to get the account number and sequence number which all `Account`s must have.

Reply:
OK. Let's keep the current way then.